### PR TITLE
Update sl-jdbc.md

### PR DESCRIPTION
**Pull request by rpourzand**
Draft PR

We now allow Dimension() to be used in a where clause for a time dimension (as opposed to TimeDimension ()). This matches the GROUP BY syntax.

We now have a way to use Order By when you are ordering by a dimension you operated on.

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/docs/dbt-cloud-apis/sl-jdbc.md
Comment (DevonFulcher) [inline website/docs/docs/dbt-cloud-apis/sl-jdbc.md]: I don't think `strongly typed` is the best way to put it. `Dimension('metric_time').grain('week')` has similar typing.

File: website/docs/docs/dbt-cloud-apis/sl-jdbc.md
Comment (mirnawong1) [inline website/docs/docs/dbt-cloud-apis/sl-jdbc.md]:
```suggestion
When using `order_by` and making changes to the object (like changing granularity), use this syntax:
```

File: website/docs/docs/dbt-cloud-apis/sl-jdbc.md
Comment (rpourzand) [reply website/docs/docs/dbt-cloud-apis/sl-jdbc.md]: Good call. Maybe "more explicit" is better? TimeDimension is more explicit than Dimension

Comment (DevonFulcher) [reply website/docs/docs/dbt-cloud-apis/sl-jdbc.md]: Ya, that is better!

File: website/docs/docs/dbt-cloud-apis/sl-jdbc.md
Comment (mirnawong1) [inline website/docs/docs/dbt-cloud-apis/sl-jdbc.md]:
```suggestion
- `Entity()` - Used for entities like primary and foreign keys - `Entity('order_id')`
```

**Review by mirnawong1 — APPROVED** [review]
hey @rpourzand , lgtm!

**Comment by mirnawong1** [comment]
hey ya'll happy to merge this if this is good to go?

**Comment by mirnawong1** [comment]
Don't merge yet per Roxi

**Review by DevonFulcher — APPROVED** [review]

File: website/snippets/_sl-partner-links.md
Comment (DevonFulcher) [inline website/snippets/_sl-partner-links.md]:
```suggestion
   - Uses Arrow Flight SQL JDBC driver version 12.0.0 or higher.
```
I would remove the Dremio part. While they also use AFS, we really have two separate products.

**Pull request merged**
